### PR TITLE
Feature/change design 6 40590157

### PR DESCRIPTION
**Pull request by sanjanadhikari1**
- [ ] Update UIs according to design changes.
- [ ] Develop LineChart component and added it to the storybook

[https://freedcamp.com/view/2956183/tasks/40590157](url)

### CODE REVIEW

**Review by ngvtuan — COMMENTED** [review]
ui issue 1
ui issue 2
ui issue 3
ui issue 4

File: package.json [hunk 1]
```diff
@@ -43,6 +43,7 @@
     "@reduxjs/toolkit": "1.5.1",
     "@storybook/addon-postcss": "2.0.0",
     "@tailwindcss/postcss7-compat": "2.1.2",
+    "@types/d3": "7.0.0",
```
[the remark this answers is not in the document]
Comment (sanjanadhikari1) [reply package.json:46]: fixed

File: package.json [hunk 2]
```diff
@@ -54,23 +55,25 @@
...
     "history": "4.10.1",
     "http": "0.0.1-security",
     "ini": "2.0.0",
-    "faker": "5.5.3",
     "javascript-terminal": "1.1.1",
     "jpg-glitch": "pastelnetwork/jpg-glitch#53a0839",
     "js-base64": "3.5.2",
     "js-sha256": "0.9.0",
     "libsodium-wrappers-sumo": "0.7.6",
+    "moment": "^2.29.1",
```
[the remark this answers is not in the document]
Comment (sanjanadhikari1) [reply package.json:76]: fixed

File: src/common/components/LineChart/LineChart.css [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  stroke-dasharray: 2, 4;
+}
+
+.tooltip {
+  fill: #141416;
+  fill-opacity: 0.88;
+}
+
+.tooltip-value {
+  stroke: white;
```
[the remark this answers is not in the document]
Comment (sanjanadhikari1) [reply src/common/components/LineChart/LineChart.css:24]: fixed

File: src/common/components/LineChart/LineChart.tsx [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+                .x(function (d) {
+                  return x(d.date)
+                })
+                .y(function (d) {
+                  return y(d.value)
+                })
+                .curve(d3.curveCardinal),
+            )
+        }
+        const focus = svg
```
[the remark this answers is not in the document]
Comment (sanjanadhikari1) [reply src/common/components/LineChart/LineChart.tsx:125]: fixed

File: src/features/nft/nftModals/ViewsStatsModal.tsx [hunk 5]
```diff
@@ -0,0 +1,179 @@
+import React, { useState } from 'react'
```
[the remark this answers is not in the document]
Comment (sanjanadhikari1) [reply src/features/nft/nftModals/ViewsStatsModal.tsx:1]: Fixed

File: src/common/components/LineChart/LineChart.tsx [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+          .on('mouseout', function () {
+            tooltipContainer.style('display', 'none')
+          })
+          .on('mousemove', () => {
+            const x0 = x.invert(d3.pointer(event)[0])
+            const i = bisectDate(data, x0, 1)
+            const d0 = data[i - 1]
+            const d1 = data[i]
+            const d =
+              moment(x0).diff(moment(d0.date)) > moment(d1.date).diff(x0)
```
Comment (CrazyOptimist) [inline src/common/components/LineChart/LineChart.tsx:176]: NB: Can be simplified to
```
x0 - d0.date > d1.date - x0
```

**Review by CrazyOptimist — CHANGES_REQUESTED** [review]

File: src/common/components/LineChart/LineChart.tsx (hunk 6 above)
Comment (sanjanadhikari1) [reply src/common/components/LineChart/LineChart.tsx:176]: fixed

File: src/common/components/LineChart/LineChart.css [hunk 7]
```diff
@@ -0,0 +1,25 @@
+.overlay {
```
[the remark this answers is not in the document]
Comment (sanjanadhikari1) [reply src/common/components/LineChart/LineChart.css:1]: fixed

File: package.json [hunk 8]
```diff
@@ -71,6 +72,7 @@
     "js-base64": "3.5.2",
     "js-sha256": "0.9.0",
     "libsodium-wrappers-sumo": "0.7.6",
+    "moment": "2.29.1",
```
[the remark this answers is not in the document]
Comment (sanjanadhikari1) [reply package.json:75]: fixed

File: src/common/components/LineChart/LineChart.module.css [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+  stroke-width: 5px;
+}
+
+.hover-line {
+  stroke: #6f257f;
+  stroke-width: 2px;
+  stroke-dasharray: 2, 4;
+}
+
+.tooltip {
```
Comment (CrazyOptimist) [inline src/common/components/LineChart/LineChart.module.css:18]: I can find only usages of `.tooltip` class, seems like other classes are not used.
Please point where are they used or remove them if non needed

**Review by CrazyOptimist — CHANGES_REQUESTED** [review]

File: src/common/components/LineChart/LineChart.module.css (hunk 9 above)
Comment (sanjanadhikari1) [reply src/common/components/LineChart/LineChart.module.css]: fixed

**Review by CrazyOptimist — APPROVED** [review]

**Review by ngvtuan — APPROVED** [review]

**Pull request merged**